python/Ganga/GPIDev/Credentials/GridProxy.py

```diff
@@ -206,12 +204,10 @@ def chooseCommandSet(self):
         Return value: None
         """
         from Ganga.GPIDev.Lib.GangaList.GangaList import GangaList
-        if "ICommandSet" == self.command._name or isType(self.command, GangaList):
-            if self.voms:
-                self.command = self.vomsCommand
-            else:
-                self.command = self.gridCommand
-        return None
+        if self.voms:
```

Comment:
Are tests of different top level packages (Ganga,  GangaDirac, ...) not 
done in different processes? If that is the case, I fail to see why this is 
required. In the old tests we had some backend matrix test that I do not 
see any reason to migrate. Better to have the tests locally with they backends.


On 9 March 2016 11:42:46 a.m. Matt Williams <notifications@github.com> wrote:

>> @@ -206,12 +204,10 @@ def chooseCommandSet(self):
>>          Return value: None
>>          """
>>          from Ganga.GPIDev.Lib.GangaList.GangaList import GangaList
>> -        if "ICommandSet" == self.command._name or isType(self.command, 
>> GangaList):
>> -            if self.voms:
>> -                self.command = self.vomsCommand
>> -            else:
>> -                self.command = self.gridCommand
>> -        return None
>> +        if self.voms:
>
> The problem at the moment is that it's cached the first time the object is 
> created and then it can never be changed again. The problem here is that we 
> have one interface (`GridProxy`) which can act in any number of ways 
> depending on the config variables when Ganga was started. Admittedly the 
> change here is only needed for running the tests but I feel that caching 
> this is unnecessarily restrictive.
>
> Information about the path, voname and info() are all cached still.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/ganga-devs/ganga/pull/292/files#r55506643